src/distribution/index.ts

```diff
@@ -0,0 +1,130 @@
+import { StandardMerkleTree } from '@openzeppelin/merkle-tree';
+import { encodeFunctionData } from 'viem';
+import {
+  REWARD_DISTRIBUTOR_ABI,
+  REWARD_DISTRIBUTOR_ADDRESSES,
+  SUPPORTED_CHAINS,
+  USERS_REWARDS,
+} from '../common/distributionData';
+
+export class RewardDistribution {
+  /**
+   * Gets the reward contract address and ABI for a specific token.
+   * @param tokenSymbol The symbol of the token.
+   * @returns The reward contract address and ABI.
+   */
+  async getRewardContract(tokenSymbol: string) {
+    tokenSymbol = tokenSymbol.toLowerCase();
+
+    const rewardContract = REWARD_DISTRIBUTOR_ADDRESSES[SUPPORTED_CHAINS.BASE][tokenSymbol];
+
+    if (!rewardContract) {
+      throw new Error(`No reward contract found for token ${tokenSymbol}`);
+    }
+
+    return {
+      address: rewardContract,
+      abi: REWARD_DISTRIBUTOR_ABI,
+    };
+  }
+
+  /**
+   * Gets the Merkle tree for a specific token and round.
+   * @param tokenSymbol The symbol of the token.
+   * @param round The round number.
+   * @returns The Merkle tree for the specified token and round.
+   */
+  async getTree(tokenSymbol: string, round: number) {
```

Comment:
The list of user rewards should be passed as a param to the function. Please update this and all the functions calling this function to accept the data for proof generation as param. The sdk should only be used to process data, which is accepted as a param from external source like the backend or the interface.